company_dependent_flag/models/base.py

```diff
@@ -26,30 +26,28 @@ def _update_company_dependent_css(self, arch):
             for field_name, field_rec in self.env[self._name]._fields.items()
             if field_rec.company_dependent
         ]
+
         for field_name in cpny_dep_fields:
             for field in arch.getElementsByTagName("field"):
                 if field.getAttribute("name") == field_name:
-                    # Create a new div element
-                    div = arch.createElement("div")
-                    div.setAttribute("class", "o_row")
+                    # Find the form parent node
+                    parent_form = field.parentNode
+                    while parent_form.tagName != "form":
+                        if parent_form.tagName == "field":
+                            break  # we are in a subview
+                        parent_form = parent_form.parentNode
 
-                    # Create a new span element
-                    span = arch.createElement("span")
-                    span.setAttribute(
-                        "class",
-                        "fa fa-lg fa-building-o",
-                    )
-                    span.setAttribute("title", "Values set here are company-specific.")
+                    if parent_form.tagName != "form":
+                        continue  # Skip this field as it's in a subview
```

Comment:
suggestion(non-blocking): this could be cleaner and more elegant if it was in its own method, IMO.